### PR TITLE
[Bundle Size][Wallet-SDK] Offload `bip32` dependency

### LINKED ISSUE #1201: [wallet-sdk] Offload `bip32` dependency
Offload `bip32` dependency from `wallet-sdk`

### PR DESCRIPTION
**Pull request by ahsan-javaid**
## Description
This PR removes `bip32` dependency from `wallet-sdk` package

Using https://github.com/paulmillr/scure-bip32 Secure, [audited](https://github.com/paulmillr/scure-bip32#security) & minimal implementation of BIP32 hierarchical deterministic (HD) wallets.

`scure-bip32` is fast and small in size very similiar to `noble-secp256k1` library.

For details refer to issue #1201

## Type of Change
- [ ] New feature
- [X] Bug fix
- [ ] API reference/documentation update
- [ ] Other

## Does this introduce a breaking change?
No

## Are documentation updates required?
No

## Testing information

1. `npm run test`

## Checklist
- [X] Code is commented where needed
- [X] Unit test coverage for new or modified code paths
- [X] `npm run test` passes
- [ ] Changelog is updated
- [X] Tag @janniks and @zone117x for review

### CODE REVIEW

File: packages/wallet-sdk/src/derive.ts [hunk 1]
```diff
@@ -248,25 +278,45 @@ export const deriveStxPrivateKey = ({
...
   index,
 }: {
-  rootNode: BIP32Interface;
+  rootNode: HDKey | BIP32Interface; // Backward compatibility with bip32
   index: number;
 }) => {
-  const childKey = rootNode.derivePath(DATA_DERIVATION_PATH).deriveHardened(index);
+  let childKey;
+  if (rootNode instanceof HDKey) {
+    childKey = rootNode.derive(DATA_DERIVATION_PATH).deriveChild(index + HIGHEST_BIT);
```
Comment (janniks) [inline packages/wallet-sdk/src/derive.ts]: ~Why do we need `HIGHEST_BIT` is some cases and not others?~

Comment (janniks) [reply packages/wallet-sdk/src/derive.ts]: ~`+ HIGHEST_BIT` "hardens", right? Should we be doing this everywhere?~

Comment (janniks) [reply packages/wallet-sdk/src/derive.ts]: Dismiss my previous comments. But let's rename `HIGHEST_BIT` to something like `HARDENED_OFFSET` to make it more clear

Comment (ahsan-javaid) [reply packages/wallet-sdk/src/derive.ts]: ✅

**Review by janniks — APPROVED** [review]

**Review by kantai — APPROVED** [review]

**Review by zone117x — APPROVED** [review]
LGTM!

**Pull request merged**